packages/next/build/swc/options.js

```diff
@@ -0,0 +1,126 @@
+const nextDistPath =
+  /(next[\\/]dist[\\/]shared[\\/]lib)|(next[\\/]dist[\\/]client)|(next[\\/]dist[\\/]pages)/
+
+function getBaseSWCOptions({
+  filename,
+  development,
+  hasReactRefresh,
+  globalWindow,
+}) {
+  const isTSFile = filename.endsWith('.ts')
+  const isTypeScript = isTSFile || filename.endsWith('.tsx')
+
+  return {
+    jsc: {
+      parser: {
+        syntax: isTypeScript ? 'typescript' : 'ecmascript',
+        dynamicImport: true,
+        // Exclude regular TypeScript files from React transformation to prevent e.g. generic parameters and angle-bracket type assertion from being interpreted as JSX tags.
+        [isTypeScript ? 'tsx' : 'jsx']: isTSFile ? false : true,
+      },
+
+      transform: {
+        react: {
+          runtime: 'automatic',
+          pragma: 'React.createElement',
+          pragmaFrag: 'React.Fragment',
+          throwIfNamespace: true,
+          development: development,
+          useBuiltins: true,
+          refresh: hasReactRefresh,
+        },
+        optimizer: {
+          simplify: false,
+          globals: {
+            typeofs: {
+              window: globalWindow ? 'object' : 'undefined',
+            },
+          },
+        },
+        regenerator: {
+          importPath: require.resolve('regenerator-runtime'),
+        },
+      },
+    },
+  }
+}
+
+export function getJestSWCOptions({ filename, esm }) {
+  let baseOptions = getBaseSWCOptions({
+    filename,
+    development: false,
+    hasReactRefresh: false,
+    globalWindow: false,
+  })
+
+  const isNextDist = nextDistPath.test(filename)
+
+  return {
+    ...baseOptions,
+    env: {
+      targets: {
+        // Targets the current version of Node.js
+        node: process.versions.node,
+      },
+    },
+    module: {
+      type: esm && !isNextDist ? 'es6' : 'commonjs',
+    },
+    disableNextSsg: true,
+    disablePageConfig: true,
```

Comment:
Do we need these for tests? If we enable the page config transform, I'm not sure how we tell the transform whether the current file is a page file since there could be many page dirs and the `findPagesDir` doesn't seem to take that into account. Plus it's probably not the best to call that for every file.

Reply:
This is fine, the removal of export const config doesn't make sense in the test env